### ISSUE TITLE
Too many environment variables

### ISSUE THREAD

**Issue by msmith-techempower** [TechEmpower/FrameworkBenchmarks#1699]
Our latest attempt at a preview run failed: http://sprunge.us/aaYM

After digging, it seems that there are too many environment variables being defined as shown in the tl;dr version here:
```
--------------------------------------------------------------------------------
  Running Test: gemini
--------------------------------------------------------------------------------

/home/techempower/FrameworkBenchmarks/config/benchmark_profile: line 49: /bin/grep: Argument list too long
/bin/bash: /usr/bin/env: Argument list too long
CRITICAL: Loading $FWROOT/config/benchmark_profile returned non-zero exit
INFO:root:Running setup module start (cwd=/home/techempower/FrameworkBenchmarks/frameworks/Java/gemini)
INFO:root:To run gemini manually, copy/paste this:
      export FWROOT=/home/techempower/FrameworkBenchmarks      && \
      export TROOT=/home/techempower/FrameworkBenchmarks/frameworks/Java/gemini       && \
      export IROOT=/home/techempower/FrameworkBenchmarks/installs       && \
      export DBHOST=10.0.3.5      && \
      export LOGDIR=/home/techempower/FrameworkBenchmarks/results/ec2/latest/logs/gemini      && \
      export MAX_THREADS=8 && \
      cd /home/techempower/FrameworkBenchmarks/frameworks/Java/gemini && \
      sudo -u testrunner -E -H stdbuf -o0 -e0 bash -ex start.sh
Setup gemini: + source /java7.installed
Setup gemini: start.sh: line 4: /java7.installed: No such file or directory
gemini: start.sh and framework processes have terminated
Setup gemini: Setup has terminated
Setup gemini: Status: Poll: 1, Port 8080 bound: False, Time Left: 1:44:59.991326
Setup gemini: start.sh process exited naturally with 1
INFO:root:Executed start.sh, returning 1
```

After doing some googlin', it seems that we have hit the limit on `RLIMIT_STACK` (8k by default) which causes exporting of additional environment variables to fail. From above, one can see that `$FWROOT` is not set properly as `start.sh` attempts `source $FWROOT/java7.installed`, but it is echoed as `source /java7.installed` because that envirnonment variable was never properly set.

**Comment by hamiltont** [comment]
As an FYI, this error doesn't normally mean you are creating too many "environment variables", it means that you are trying to pass too many arguments to a program like grep that can only handle a limited number (IIRC, quite literally the kernel buffer used to copy the argument string from the calling shell to the newly-launched program has an upper limit). You can normally run `getconf ARG_MAX` to see the max size of the argument strings. Normally this is a shell wildcard run amuk - e.g. we would expect to see a command like `do_command_foo *`, and the `*` was blowing up the argument list. 

This line from our benchmark_profile `export NUMCPUS=grep -c '^processor' /proc/cpuinfo` is printing an error, but I don't see how that could be expanding into a large argument list... 

Do you know where in the source code we are getting this output from `/bin/bash: /usr/bin/env: Argument list too long`? Would be helpful to see both source code lines that are complaining

**Comment by msmith-techempower** [comment]
Well, fine. Lol.

It's resolved by #1605 which I intended to merge today... but you started merging others so it'll have to wait a day while I reconcile. nbd
